core/controllers/collection_editor.py

```diff
@@ -196,6 +196,11 @@ def put(self, collection_id):
                     self.user_id, collection_id)
                 collection_services.index_collections_given_ids([
                     collection_id])
+            elif not is_public:
+                collection_services.unpublish_collection_update_user_profiles(
```

Comment:
Are you trying to do something like:

https://github.com/oppia/oppia/blob/dcd59cab0cf8aadeb515d0882be8d8fba6f4c9eb/core/controllers/editor.py#L428

That's how the exploration editor performs unpublishing.